nrf_rpc/include/nrf_rpc.h

```diff
@@ -171,6 +171,20 @@ struct nrf_rpc_err_report {
 	enum nrf_rpc_packet_type packet_type;
 };
 
+/** @brief Cleanup procedure list element.
+ */
+struct nrf_rpc_cleanup_handler
+{
+	/** @brief Handler to the cleanup function invoked on the call to @ref nrf_rpc_stop. */
+	void (*handler)(void *context);
+
+	/** @brief Custom context passed as the function parameter */
+	void *context;
+
+	/** @brief Pointer to the next element in the list. Managed automatically by the nrf_rpc */
+	struct nrf_rpc_cleanup_handler *next;
```

Comment:
nit: why write linked list manually when you already have `slist` type in zephyr ? 
adding/removing/iterating over such lists would be a one liner, and it does not include any overhead, as linked list is probably already used in zephyr, so no additional code need to be pulled. 

https://docs.zephyrproject.org/latest/kernel/data_structures/slist.html

Reply:
OK, I was not aware of this requirement